src/domain/event/eventForm/OrderableEventCheckboxField.tsx

```diff
@@ -0,0 +1,27 @@
+import { FieldProps } from 'formik';
+import { CheckboxProps } from 'hds-react';
+import * as React from 'react';
+
+import CheckboxField from '../../../common/components/form/fields/CheckboxField';
+
+// This is its own component because how Formik recommends to handle situations
+// where form value is based on other form value.
+// see: https://github.com/formium/formik/issues/2204#issuecomment-574207100
+const OrderableEventCheckboxField = ({
+  form,
+  ...props
+}: FieldProps & CheckboxProps) => {
+  const { value: isOrderable } = props.field;
+  const { setFieldValue } = form;
+
+  React.useEffect(() => {
+    if (isOrderable) {
+      setFieldValue('location', '');
+      setFieldValue('isVirtual', false);
+    }
+  }, [isOrderable, setFieldValue]);
```

Comment:
Wondering about this--why not override the `onChange` handler? Pseudo:

```tsx
(e) => {
  const value = e.target.value;

  if (value) {
    setFieldValue('location', '');
    setFieldValue('isVirtual', false);
  }

  setFieldValue(e.target.name, value);
}
```

Seems like the usage of `useEffect` could result in situations where this component ends up syncing state when `isOrderable` is controlled in some other component. That seems like something that could make the logic of the form harder to understand when debugging.

Reply:
I'm afraid that this is done with useEffect because your example breaks Formiks internal validation logic.

Reply:
I have had problems with validation going out of sync when trying to set multiple field values inside one event handler

Reply:
I think I tried that also when I tried to solve this problem. But not necessarily in separate component like this one...

Reply:
Good to know 👍